.travis.yml

```diff
@@ -1,34 +1,62 @@
 ---
 language: node_js
 node_js:
-  - "0.12"
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
+  - "10"
 
-sudo: false
+dist: trusty
+
+addons:
+  chrome: stable
 
 cache:
-  directories:
-    - node_modules
+  yarn: true
 
 env:
-  - EMBER_TRY_SCENARIO=default
-  - EMBER_TRY_SCENARIO=ember-release
-  - EMBER_TRY_SCENARIO=ember-beta
-  - EMBER_TRY_SCENARIO=ember-canary
+  global:
+    # See https://git.io/vdao3 for details.
+    - JOBS=1
+
+branches:
+  only:
+    - master
+    # npm version tags
+    - /^v\d+\.\d+\.\d+/
 
-matrix:
+jobs:
   fast_finish: true
-  allow_failures:
-    - env: EMBER_TRY_SCENARIO=ember-canary
+
+  include:
+    # runs tests with current locked deps
+    - stage: "Tests"
+      name: "Tests"
+      script:
+        - yarn test:ember
+
+    - stage: "Additional Tests"
+      name: "Floating Dependencies"
+      install:
+        - yarn install --no-lockfile --non-interactive
+        - bower install
+      script:
+        - yarn test:ember
+
+    # we recommend new addons test the current and previous LTS
+    # as well as latest stable release (bonus points to beta/canary)
+    - env: EMBER_TRY_SCENARIO=ember-1.13-with-ember-legacy-views
+    - env: EMBER_TRY_SCENARIO=ember-1.13-without-ember-legacy-views
+    - env: EMBER_TRY_SCENARIO=ember-2.5-with-ember-legacy-views
+    - env: EMBER_TRY_SCENARIO=ember-2.5-without-ember-legacy-views
```

Comment:
Let's add 2.4 as well (since I think in general, it would be better for folks to be on 2.4 than 2.5 since it was an LTS).